### PR TITLE
Improve showing idf project templates readme format in the new project wizard

### PR DESCRIPTION
**Pull request by sigmaaa**
Markdown text from readme.md is converting to HTML format and then setting in the new swt browser widget, which is added instead of ScrolledFormText. This allows seeing the content clearly with all indentation and rich text features.

Also was added locationListener to new browser component, which prevents open HTTP hyperlink in a browser component, so URL will be opened in default external browser
IEP-162

### CODE REVIEW

File: bundles/com.espressif.idf.ui/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -29,10 +29,19 @@ Require-Bundle: org.eclipse.core.runtime,
...
- com.espressif.idf.ui.dialogs
+ com.espressif.idf.ui.dialogs,
+ com.espressif.idf.ui.handlers,
+ com.espressif.idf.ui.help,
+ com.espressif.idf.ui.install,
+ com.espressif.idf.ui.preferences,
+ com.espressif.idf.ui.size,
+ com.espressif.idf.ui.templates,
+ com.espressif.idf.ui.update,
+ com.espressif.idf.ui.wizard
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/META-INF/MANIFEST.MF:45]: I think we can remove them if it's not used anywhere else

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java [hunk 2]
```diff
@@ -36,15 +46,42 @@ public void createControl(Composite parent) {
...
+					{
+						PlatformUI.getWorkbench().getBrowserSupport().getExternalBrowser()
+								.openURL(new URL(event.location));
+						event.doit = false;
+					}
+					catch (
+							PartInitException
+							| MalformedURLException e)
+					{
+						e.printStackTrace();
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java:66]: Log the exception

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java [hunk 3]
```diff
@@ -64,17 +101,26 @@ public Control getControl() {
 		return container;
 	}
 
-	public void setText(String text) {
-		this.text = text;
+	public void setText(String text)
+	{
+		this.text = convertMarkdownToHtml(text);
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java:106]: We can move this below to include within the condition if (formText != null) {...}

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java [hunk 4]
```diff
@@ -64,17 +101,26 @@ public Control getControl() {
...
 				Status status = new Status(IStatus.ERROR, IDFCorePlugin.PLUGIN_ID, "Error parsing the template description", e); //$NON-NLS-1$
 				IDFCorePlugin.getPlugin().getLog().log(status);
 			}
 	}
+
+	private String convertMarkdownToHtml(String markdownText)
+	{
+		Parser parser = Parser.builder().build();
+		Node document = parser.parse(markdownText);
+		HtmlRenderer renderer = HtmlRenderer.builder().build();
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java:123]: Add exception block, if something goes wrong with the markdown parser it should still need to display the original text

**Review by kolipakakondal — CHANGES_REQUESTED** [review]
LGTM. Could you please address the comments mentioned?

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java (hunk 2 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java]: @kolipakakondal 
I used  IDF Eclipse plug-in logger:
https://github.com/espressif/idf-eclipse-plugin/blob/db760d63e4174b294d5c8753cebe7ffced8ccdc1/bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java#L62-L67

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java (hunk 3 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java]: @kolipakakondal 
done:
https://github.com/espressif/idf-eclipse-plugin/blob/db760d63e4174b294d5c8753cebe7ffced8ccdc1/bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java#L104-L111

File: bundles/com.espressif.idf.ui/META-INF/MANIFEST.MF (hunk 1 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/META-INF/MANIFEST.MF]: I removed them. It was added automatically and I forgot to remove it. Sorry for that

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java (hunk 4 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java]: @kolipakakondal 
the try-catch block was added, exception is writing by Logger. The default text is returned when something goes wrong:
https://github.com/espressif/idf-eclipse-plugin/blob/db760d63e4174b294d5c8753cebe7ffced8ccdc1/bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/templates/FormBrowser.java#L119-L134

**Comment by kolipakakondal** [comment]
LGTM

**Review by kolipakakondal — APPROVED** [review]

**Pull request merged**